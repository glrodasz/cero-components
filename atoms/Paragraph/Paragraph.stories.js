import { Paragraph, options, styles } from '.'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(Paragraph, styles)
const ListTemplate = getListTemplate(Paragraph, styles)

export default {
  title: 'Atoms/Paragraph',
  component: Paragraph,
  args: {
    children:
      'Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange UdÃ¹n piled fighters potent?',
  },
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
    weight: getOptionsArgTypes(options.weights),
    children: { control: 'text' },
  },
}

export const Default = Template.bind({})

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }

export const Weights = ListTemplate.bind({})
Weights.args = { items: options.weights.map((weight) => ({ weight })) }
