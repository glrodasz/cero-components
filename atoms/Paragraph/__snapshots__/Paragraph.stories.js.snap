// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Paragraph Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="paragraph color-base size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
  <div
    className="paragraph color-muted size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
  <div
    className="paragraph color-inverted size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
</div>
`;

exports[`Storyshots Atoms/Paragraph Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="paragraph color-base size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
</div>
`;

exports[`Storyshots Atoms/Paragraph Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="paragraph color-base size-sm weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
  <div
    className="paragraph color-base size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
  <div
    className="paragraph color-base size-lg weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
</div>
`;

exports[`Storyshots Atoms/Paragraph Weights 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="paragraph color-base size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
  <div
    className="paragraph color-base size-md weight-medium"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </div>
</div>
`;
